### DISCUSSION TITLE
How to configure beforeEnter in Per-Route Guards

### DISCUSSION THREAD

**Discussion by JaavierR** [antfu/vitesse#290] [Q&A]
Hi, 
I was testing the template as I am interested in using it on some projects as starting point. But I can't manage to add the beforeEnter hook at route level, by this I mean:

```js

{
  path: '/',
  ...,
  beforeEnter: () => {}
}
```

When using extendRoute I am getting the following error.

```bash
  1  |  import _src_pages_index_vue from "/src/pages/index.vue";
     |  ^
  2  |  
  3  |  const routes = [{"name":"index","path":"/","component":_src_pages_index_vue,"props":true,"meta":{"auth":true}},{"name":"Error","path":"/:all(.*)*","component":() => import('/src/pages/[...all].vue'),"props":true,"beforeEnter":(to)=>{console.log(\Thisisnew\",to.params);}"}];
```

And this is the current setup in the extendRoute options:

```js
extendRoute(route, parent) {
if (route.name === 'Test') {
   return {
           ...route,
           beforeEnter: (to) => {
           console.log('This is new', to.params)
           },
      }
}}
```
I don't know if someone already manage to implement this. And also I searched in the repository of vite-plugin-page but i didn't found anything related to, and also ask in the discussion but I don't have any response so I tried here if someone is trying to implement something similar. So I ask here in the hope that someone here has encountered the same problem.

**Comment by hannoeru** [comment]
Because it's difficult to serialize the function then pass it to frontend, I suggest you modify the routes directly from `~pages` import in `main.ts`.
